c10/test/core/DispatchKeySet_test.cpp

```diff
@@ -370,29 +367,12 @@ TEST(DispatchKeySet, IteratorCrossProduct) {
 
 TEST(DispatchKeySet, IteratorFull) {
   DispatchKeySet full_set(DispatchKeySet::FULL);
-  uint8_t i = 0;
-
-  for (const auto& it : full_set) {
-    i++;
-  }
-  // Total # of runtime entries includes an entry for DispatchKey::Undefined,
-  // which is not included when iterating through the DispatchKeySet.
-  ASSERT_EQ(i, num_runtime_entries - 1);
-}
-
-TEST(DispatchKeySet, IteratorRangeFull) {
-  DispatchKeySet full_set(DispatchKeySet::FULL);
-  uint8_t i = 0;
-
-  for (DispatchKey dispatch_key : full_set) {
-    i++;
-  }
+  std::ptrdiff_t count = std::distance(full_set.begin(), full_set.end());
```

Comment:
Is this doing the same thing?
The goal here is really to go through the whole iterator.
Also the goal was to explicitely test that both iteration types are working.

cc @bdhirsh

Reply:
I think it must. It's not a random access iterator so distance must advance one at a time until it reaches the end.